File: src/Web/WebAngular/src/app/domain/view-models/connector-function-detail.view-model.ts
import { BuildStatus } from "../enums/build-status.enum";

export interface ConnectorFunctionDetailViewModel {
	id: string;
	name: string;
	function: string;
	friendlyName: string;
	connector: string
	version: string;
	specsFile: string;
	script: string;
	package: string;
	buildStatus: BuildStatus;
	buildStderr: string | null;
	createdBy: string
	creationDate: string;
	updatedBy: string;
	lastUpdate: string;
}

File: src/Core/Houston.Application/ViewModel/ConnectorFunctionViewModels/ConnectorFunctionDetailViewModel.cs
namespace Houston.Application.ViewModel.ConnectorFunctionViewModels {
	public class ConnectorFunctionDetailViewModel {
		public Guid Id { get; set; }
		
		public string Name { get; set; } = null!;

		public string FriendlyName { get; set; } = null!;

		public string Connector { get; set; } = null!;

		public string Version { get; set; } = null!;

		public byte[] Script { get; set; } = null!;

		public byte[] Package { get; set; } = null!;

		public byte[] SpecsFile { get; set; } = null!;

		public BuildStatus BuildStatus { get; set; }

		public byte[]? BuildStderr { get; set; }

		public string CreatedBy { get; set; } = null!;

		public DateTime CreationDate { get; set; }

		public string UpdatedBy { get; set; } = null!;

		public DateTime LastUpdate { get; set; }
	}
}
